packages/shared/lib/services/sync/data/records.service.ts

```diff
@@ -290,6 +326,168 @@ export async function getDataRecords(
+                        this.where('created_at', '=', db.knex.raw('updated_at')).orWhere('created_at', '!=', db.knex.raw('updated_at'));
+                    });
+                    break;
+                case filter.includes('updated') && filter.includes('deleted'):
+                    query = query.andWhere(function () {
+                        this.where('external_is_deleted', true).orWhere('external_deleted_at', null).andWhere('created_at', '!=', db.knex.raw('updated_at'));
+                    });
+                    break;
+                case filter.includes('added') && filter.includes('deleted'):
+                    query = query.andWhere(function () {
+                        this.where('external_is_deleted', true).orWhere('external_deleted_at', null).andWhere('created_at', '=', db.knex.raw('updated_at'));
+                    });
+                    break;
+                case filter === 'added':
+                    query = query.andWhere('external_deleted_at', null).andWhere('created_at', '=', db.knex.raw('updated_at'));
+                    break;
+                case filter === 'updated':
+                    query = query.andWhere('external_deleted_at', null).andWhere('created_at', '!=', db.knex.raw('updated_at'));
+                    break;
+                case filter === 'deleted':
+                    query = query.andWhere({ external_is_deleted: true });
+                    break;
+            }
+        }
+
+        let nextCursor = null;
+
+        const rawResult = await query.select(
+            'id',
+            db.knex.raw(`
+                jsonb_set(
+                    json::jsonb,
+                    '{_nango_metadata}',
+                    jsonb_build_object(
+                        'first_seen_at', created_at,
+                        'last_modified_at', updated_at,
+                        'deleted_at', external_deleted_at,
+                        'last_action',
+                        CASE
+                            WHEN external_deleted_at IS NOT NULL THEN 'DELETED'
+                            WHEN created_at = updated_at THEN 'ADDED'
+                            ELSE 'UPDATED'
+                        END
+                    )
+                ) as record
+            `)
+        );
+
+        const result = encryptionManager.decryptDataRecords(rawResult, 'record') as unknown as SyncDataRecord[];
+
+        if (result.length === 0) {
+            return { success: true, error: null, response: { records: [], next_cursor: nextCursor } };
+        }
+
+        const customerResult = result.map((item) => item.record);
+
+        if (customerResult.length > (limit || 100)) {
+            customerResult.pop();
+            rawResult.pop();
+
+            const cursorRawElement = rawResult[rawResult.length - 1] as SyncDataRecord;
+            const cursorElement = customerResult[customerResult.length - 1] as unknown as CustomerFacingDataRecord;
```

Comment:
do I understand correctly the cursor is a pointer to the last records that is gonna be sent back?

Reply:
That is correct, and the next page query looks for records  > the provided cursor

Reply:
I am nitpicking so you do as you prefer. `next` could actually be the first record following the last one of a page which would be more in phase with the name `next`. The condition would then be `>= provided cursor`.
Not big deal though if you prefer to keep it this way

Reply:
Yeah, I see your point. Think I'll keep it like this for now.